sim/src/main/cc/bridges/TracerVModule.cc

```diff
@@ -0,0 +1,367 @@
+// See LICENSE for license details.
+
+#include "bridges/peek_poke.h"
+#include "bridges/tracerv.h"
+#include "core/bridge_driver.h"
+#include "core/simif.h"
+#include "core/simulation.h"
+
+#include <algorithm>
+#include <iostream>
+#include <string_view>
+
+static std::vector<bool> get_contiguous(const unsigned bits,
+                                        const unsigned total) {
+  std::vector<bool> ret;
+  for (unsigned i = 0; i < total; i++) {
+    const bool value = (i < bits);
+    ret.emplace_back(value);
+  }
+
+  return ret;
+}
+
+static std::vector<uint64_t> get_iaddrs(const unsigned step,
+                                        const unsigned total) {
+  constexpr uint64_t offset =
+      1024; // should be larger than total but doesn't really matter
+  std::vector<uint64_t> ret;
+
+  for (unsigned i = 0; i < total; i++) {
+    ret.emplace_back(step * offset + i);
+  }
+
+  return ret;
+}
+
+static std::string namei(const unsigned x) {
+  std::stringstream ss;
+  ss << "io_insns_" << x << "_iaddr";
+  return ss.str();
+};
+
+static std::string nameinsn(const unsigned x) {
+  std::stringstream ss;
+  ss << "io_insns_" << x << "_insn";
+  return ss.str();
+};
+
+static std::string namev(const unsigned x) {
+  std::stringstream ss;
+  ss << "io_insns_" << x << "_valid";
+  return ss.str();
+};
+
+// a simple way to turn the iaddr values we use
+// in this test into a 5 bit value
+static uint32_t fold(const uint64_t x) {
```

Comment:
Could you rename the argument/locals so that the `iaddr` in the comment is somewhat tied to the source?